src/java/main/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -199,14 +199,16 @@ public synchronized boolean append(TxnHeader hdr, Record txn)
                 LOG.warn("Current zxid " + hdr.getZxid()
```

Comment:
I would recommend make the early return changes here as the changes to be made is small and they are not hard to review. When we say "do it separately" in the context of zk, it probably implies "never" <grin>